src/python/gudhi/representations/vector_methods.py

```diff
@@ -510,24 +510,27 @@ def transform(self, X):
         for i in range(num_diag):
             orig_diagram, diagram, num_pts_in_diag = X[i], new_X[i], X[i].shape[0]
             try:
-                new_diagram = DiagramScaler(use=True, scalers=[([1], MaxAbsScaler())]).fit_transform([diagram])[0]
+                #new_diagram = DiagramScaler(use=True, scalers=[([1], MaxAbsScaler())]).fit_transform([diagram])[0]
+                new_diagram = DiagramScaler().fit_transform([diagram])[0]
             except ValueError:
                 # Empty persistence diagram case - https://github.com/GUDHI/gudhi-devel/issues/507
                 assert len(diagram) == 0
                 new_diagram = np.empty(shape = [0, 2])
-
+                
+            p = new_diagram[:,1]
+            L = sum(p)
+            p = p/L
```

Comment:
I'd rather avoid the generic `sum` and use numpy's version, so `p=p/np.sum(p)` (no need to name `L` if it is only used once), or a version with `np.linalg.norm`.

Reply:
I agree p=p/np.sum(p) is better